pkg/providers/providersutils.go

```diff
@@ -277,8 +282,34 @@ func GetCurrentAzSubscriptionId() []string {
 		log.Fatal(err)
 	}
 
-	var ids []string
-	json.Unmarshal(out, &ids)
+	var id []string
+	json.Unmarshal(out, &id)
+
+	return id[0]
+}
+
+func GetAzSubscriptionLabels(id string) []SubLabel {
+	CheckAzCliInstalled()
+	if !IsLoggedInToAz() {
+		if err := LogInToAz(); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	getAccountCmd := exec.Command("az", "account", "list", "--all", "--query", "[].{id: id, name: name}")
+	out, err := getAccountCmd.CombinedOutput()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var subLabels []SubLabel
```

Comment:
💡 Purely by choice: but why not **`break`** once condition is met, or is there a hidden special case.

```*.go
        var subLabels []SubLabel
	if err := json.Unmarshal(out, &subLabels); err != nil {
		log.Fatalf("Failed to unmarshal JSON output: %v", err)
	}

	// Reorder the subLabels to move the one with the matching ID to the front
	for idx, val := range subLabels {
		if val.ID == id {
			subLabels[0], subLabels[idx] = subLabels[idx], subLabels[0]
			break
		}
	}
```